### ISSUE TITLE
LGM crashes on startup (on Mac) trying to communicate with the library seems partially successful

### ISSUE THREAD

**Issue by time-killer-games** [enigma-dev/enigma-dev#1733]
terminal contents after crash:

https://pastebin.com/UNVUCZab

**Comment by RobertBColton** [comment]
The Java version may be too new. LGM's minimum Java version is 1.7. It is also possible that you need to increase the maximum JVM heap size using the command line parameter. Other than that, this might not even be a LateralGM issue since I see JDI crapping out a bunch.

**Comment by time-killer-games** [comment]
@RobertBColton i updated the issue with more details

**Comment by RobertBColton** [comment]
You can alias make in the compiler eys:
https://github.com/enigma-dev/enigma-dev/blob/master/Compilers/MacOSX/gcc.ey

**Comment by time-killer-games** [comment]
ok its giving me less stuff in the errors now but it still crashes

**Comment by time-killer-games** [comment]
how do i et the default platform to sdl? right now its trying to use cocoa which might be the issue idk

**Comment by RobertBColton** [comment]
Make Cocoa represents "None" I think or just delete the "Info" folder altogether.
https://github.com/enigma-dev/enigma-dev/blob/148023a668ad72927fc165bc0bc1c24952e5077e/ENIGMAsystem/SHELL/Platforms/Cocoa/Info/About.ey#L11

**Comment by time-killer-games** [comment]
ok after setting the default to SDL im getting this:

https://pastebin.com/F7G1mQmg

**Comment by RobertBColton** [comment]
You didn't increase the heap size.

**Comment by time-killer-games** [comment]
ill try downgrading next, but the max and min heap size no longer appears to be located in settings.ey anymore i have no idea where you guys have moved it to

**Comment by RobertBColton** [comment]
The min and max heap parameters are a command line parameter to Java.

**Comment by time-killer-games** [comment]
anyways that used to be in the settings.ey and i think it should still be for convenience tbh

**Comment by RobertBColton** [comment]
No, it was in `settings.ini` where it was added by me to the Portable ZIP.

**Comment by time-killer-games** [comment]
ah. yeah well what min and max would you recommend? nothing seems to be working, only crash is getting slightly delayed. Is openjdk or oracle the one we need on mac?

**Comment by time-killer-games** [comment]
i can't figure out how to downgrade so ill just have to try that a different day

**Comment by RobertBColton** [comment]
Oracle JDK is always recommended on every platform.
https://enigma-dev.org/docs/Wiki/Troubleshoot#OutOfMemoryError

**Comment by fundies** [comment]
whatever nonsense shell your using is missing a bunch of bash commands like which and mkdir. It's not going to work unless you're in a proper bash shell

**Comment by time-killer-games** [comment]
@fundies I'm using the default Mac terminal, the same thing I used when you helped me get it working last year. Also, more people have complained about me on discord than you from what I can tell, YellowAfterlife even asked if he could block me recently because I was so annoying to him. I think I might go back to using just hangouts soon, if this continues, I think Josh really shouldn't have booted you out, even if it was meant to temporary afaik,

**Comment by time-killer-games** [comment]
@RobertBColton fucking apple replaced instances I typed gmake with make so anyway now that's corrected the original post of this issue now makes proper sense.

**Comment by time-killer-games** [comment]
anyway I'm ready to work on this gain. I'll do what robert suggested and look at that specific commit he mentioned to me on discord.

**Comment by fundies** [comment]
> make[1]: which: Command not found
> make[2]: uname: Command not found
> make[2]: which: Command not found
> make[2]: mkdir: No such file or directory

you need to solve that first

**Comment by time-killer-games** [comment]
@fundies fixed it. I forgot to use gmake instead of make.

Now I'm getting this:

OptionsParser.cpp:215:28: error: out-of-line definition of 'APIyaml' does not
      match any declaration in 'OptionsParser'
std::string OptionsParser::APIyaml()

**Comment by fundies** [comment]
im guessing you messed up that source file somehow.

**Comment by time-killer-games** [comment]
@fundies normally i would say your first statement is probably correct, but I've uninstalled enigma and started over several times now and this is one of multiple times i was stopped by this same error at this point in time, at least twice in a row now.

I'll check to see what else is mentioned in the terminal later and pastebin the results to you.

**Comment by fundies** [comment]
std::string OptionsParser::APIyaml() takes an argument in the latest sources so either its edited or theres something more to that error

**Comment by time-killer-games** [comment]
Yeah, that's what I'm saying. I don't really know for sure, but I think the issue lies somewhere else in the other contents printed to the terminal. In any case, I just ran gmake clean and gmake all again. I'll post the contents of the terminal once it reaches that error again along with the current contents of my copy of std::string OptionsParser::APIyaml() as separate pastebin's, and hopefully we'll get to the bottom of it.

**Comment by time-killer-games** [comment]
Here's what the terminal printed for gmake all: https://pastebin.com/EG84BhSk
Here's the contents of my current copy of OptionsParser::APIyaml(): https://pastebin.com/9v92Br0Y

**Comment by fundies** [comment]
and as i said https://github.com/enigma-dev/enigma-dev/blob/master/CommandLine/emake/OptionsParser.cpp#L216 your line 215 doesnt match at all. youre clearly on edited source

**Comment by time-killer-games** [comment]
Thanks for that. I didn't realize i edited anything, or maybe i copied from a wrong commit. However it happened, that part should be fixed now. Thanks for the help!

**Comment by time-killer-games** [comment]
Ok, so emake built it appears even though building the Cocoa test harness failed but i don't think that makes a difference because i don't think we ever got that working iirc.

now trying to build with emake gives me this:

`Last login: Wed May 29 17:01:32 on ttys000
Owners-MacBook-Pro:~ owner$ cd /Users/owner/enigma-dev 
Owners-MacBook-Pro:enigma-dev owner$ ./emake "/Users/owner/Desktop/Key to Success.gmk" -x gcc -p SDL -c Precise -g OpenGL1 -a OpenAL -w None -e "Paths,Alarms" -o "/Users/owner/Desktop/Key to Success" -r
[libprotobuf ERROR google/protobuf/descriptor_database.cc:58] File already exists in database: options.proto
[libprotobuf FATAL google/protobuf/descriptor.cc:1370] CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size): 
libc++abi.dylib: terminating with uncaught exception of type google::protobuf::FatalException: CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size): 
Abort trap: 6
Owners-MacBook-Pro:enigma-dev owner$`

**Comment by fundies** [comment]
looks like libprotocols or protobuf is being linked twice and google dont like it

**Comment by time-killer-games** [comment]
what do you mean linked twice? As in the the libProtocols.dylib? Do you have any idea how to correct this, or what file is most like in need of changing? Is there any particular file you need a pastebin of?

**Comment by fundies** [comment]
you need to ensure the dylibs your building and linking are shared and not static

**Comment by fundies** [comment]
anyways, I'm done giving hints. I'll sell you my top secret mac formula for $200usd

**Comment by time-killer-games** [comment]
@fundies thanks for the help you already provided though, I never would've made it this far without you. You're very appreciated by me, and I feel you aren't given enough credit by Josh. He said "fundies served as a good motivator for me and Robert, but other than that, he mostly just a pain in the ass". I find Josh's attitude towards you unrealistic, inaccurate, and absurd. You have always been one of the most helpful members we've ever had, and other than Robert, probably the biggest contributor to the project as well.

I'm sorry Josh let go of you like this, and I wish you the best on finding another project to work on that will be just as if not much more fulfilling for you, in a community that will be a lot more appreciative of you. It's been a rollercoaster working with you fundies, but it has been an honor all the same.

Thank you for your years of dedication and putting up with us.

**Comment by RobertBColton** [comment]
@time-killer-games If this has been resolved, please close it. You've been using ENIGMA on Mac so I suspect it's solved.

**Comment by time-killer-games** [comment]
@RobertBColton I haven't been using enigma on Mac. My ENIGMA extensions were tested in Xcode projects with all unrelated code to the extension in enigma taken out.
